vm/src/stdlib/posix.rs

```diff
@@ -687,21 +687,54 @@ pub mod module {
 
     #[pyfunction]
     fn chmod(
-        path: PyPathLike,
+        path: PathOrFd,
         dir_fd: DirFd<0>,
         mode: u32,
         follow_symlinks: FollowSymlinks,
         vm: &VirtualMachine,
     ) -> PyResult<()> {
         let [] = dir_fd.0;
-        let body = move || {
-            use std::os::unix::fs::PermissionsExt;
-            let meta = fs_metadata(&path, follow_symlinks.0)?;
-            let mut permissions = meta.permissions();
-            permissions.set_mode(mode);
-            fs::set_permissions(&path, permissions)
-        };
-        body().map_err(|err| err.into_pyexception(vm))
+        match path {
+            PathOrFd::Path(path) => {
+                let body = move || {
+                    use std::os::unix::fs::PermissionsExt;
+                    let meta = fs_metadata(&path, follow_symlinks.0)?;
+                    let mut permissions = meta.permissions();
+                    permissions.set_mode(mode);
+                    fs::set_permissions(&path, permissions)
+                };
+                body().map_err(|err| err.into_pyexception(vm))
+            }
+            PathOrFd::Fd(fd) => nix::sys::stat::fchmod(
+                fd,
+                nix::sys::stat::Mode::from_bits(mode as libc::mode_t).unwrap(),
+            )
+            .map_err(|err| err.into_pyexception(vm)),
+        }
```

Comment:
Because `chmod` is superset of `fchmod` and `lchmod`,  I think `chmod` calling `fchmod` and `lchmod` is more natural way.
It seems you decided to do this way because `chmod` is not fully consistable with python `fchmod` and `lchmod` because they are not the full feature due to lack of `FollowSymlinks`. In such case, I still think adding `_fchmod` and `_lchmod` with `FollowSymlinks` and using them for all `chmod`, `fchmod` and `lchmod` is better.

Even though I think this not a common design, I still can accept this approach because it have a reasonable excuse of its reversed design flow. But because this is counterintuitive decision, I think the decision itself must be documented enough not to confuse future contributors to think this is a common approach in this project.

Reply:
OK. I changed these.